frontend/src/app/pages/post-battle-page.tsx

```diff
@@ -1,30 +1,29 @@
+import React from 'react';
 import { BattleResultResponse } from '@game/shared';
 import { getPostBattleData } from '../api/get-post-battle-data';
 import { Card, Center, Loader, SimpleGrid } from '@mantine/core';
 import { TeamInfo } from '../components/post-battle-page/team-info';
 import { useQuery } from 'react-query';
 
 export const PostBattlePage = () => {
-
-  const { isLoading,isError, data: battleResult  } = useQuery(['battle-result'], (): Promise<BattleResultResponse> => getPostBattleData('http://localhost:3333/api/battle/1/result'));
-
-  const errorMessage = "An error occurred.";
-  const bResult = "Battle result";
+  const apiUrl = process.env.LOCAL_URL || 'http://localhost:3333/api/battle/1/result';
+  const { isLoading, isError, error, data: battleResult  } = useQuery(['battle-result'], (): Promise<BattleResultResponse> => getPostBattleData(apiUrl));
 
   if (isError) {
-    return <div>{errorMessage}</div>;
+    console.error('Error fetching battle result:', error);
+    return <div>An error occurred.</div>;
   }
   if (isLoading) {
     return <Center><Loader color={'white'} /></Center>;
   }
-  const sortedWinners = battleResult?.winners.sort((a, b) => b.score - a.score) || [];
-  const sortedLosers = battleResult?.losers.sort((a, b) => b.score - a.score) || [];
+  const sortedWinners = React.useMemo(() => battleResult?.winners.sort((a, b) => b.score - a.score) || [], [battleResult]);
+  const sortedLosers = React.useMemo(() => battleResult?.losers.sort((a, b) => b.score - a.score) || [], [battleResult]);;
 
   return (
     <Card shadow="sm" title={'Post Battle Page'} withBorder>
       <Card.Section>
         <Center>
-          <h1>{bResult}</h1>
+          <h1>Battle result</h1>
         </Center>
       </Card.Section>
       <SimpleGrid spacing="md" cols={2}>
```

Comment:
1. **Environment Variable Usage**: The `apiUrl` is set using `process.env.LOCAL_URL`, which may expose sensitive information if not properly managed. Ensure that environment variables are securely configured and not exposed in client-side code.

2. **Error Handling**: While logging errors with `console.error` is useful for debugging, consider implementing a user-friendly error message or retry mechanism to improve user experience.

3. **Performance Optimization**: The use of `React.useMemo` for sorting winners and losers is appropriate; however, ensure that the `battleResult` object is immutable or deeply compared to avoid unnecessary recalculations.

4. **Security Risk**: Hardcoding URLs like `'http://localhost:3333/api/battle/1/result'` can lead to security vulnerabilities. Consider using configuration files or environment variables to manage API endpoints securely.

Quality Rate: 6/10